src/core/operations_problem.jl

```diff
@@ -273,6 +281,76 @@ get_output_dir(problem::OperationsProblem) = problem.internal.output_dir
 get_variables(problem::OperationsProblem) = get_optimization_container(problem).variables
 get_run_status(problem::OperationsProblem) = problem.internal.run_status
 set_run_status!(problem::OperationsProblem, status) = problem.internal.run_status = status
+get_time_series_cache(problem::OperationsProblem) = problem.internal.time_series_cache
+empty_time_series_cache!(x::OperationsProblem) = empty!(get_time_series_cache(x))
+
+function get_time_series_array!(
+    ts_type::Type{<:IS.TimeSeriesData},
+    problem::OperationsProblem,
+    component,
+    name,
+    initial_time;
+    ignore_scaling_factors = true,
+)
+    cache = get_time_series_cache(problem)
+    key = TimeSeriesCacheKey(IS.get_uuid(component), ts_type, name)
+    if haskey(cache, key)
+        ts_cache = cache[key]
+    else
+        ts_cache = make_time_series_cache(
+            ts_type,
+            component,
+            name,
+            initial_time,
+            ignore_scaling_factors = true,
+        )
+        cache[key] = ts_cache
+    end
+
+    ts = IS.get_next_time_series_array!(ts_cache)
+    @assert_op IS.get_initial_timestamp(ts) == initial_time
+    return ts
+end
+
+function make_time_series_cache(
+    time_series_type::Type{T},
+    component,
+    name,
+    initial_time;
+    ignore_scaling_factors = true,
+) where {T <: IS.TimeSeriesData}
+    key = TimeSeriesCacheKey(IS.get_uuid(component), T, name)
+    if T <: IS.SingleTimeSeries
+        cache = IS.StaticTimeSeriesCache(
+            PSY.SingleTimeSeries,
+            component,
+            name,
+            start_time = initial_time,
+            ignore_scaling_factors = ignore_scaling_factors,
+        )
+    elseif T <: IS.Deterministic
+        cache = IS.ForecastCache(
```

Comment:
There are customizations we could make here.
- IS allows custom cache sizes per array.
- We could disable the feature entirely with a flag. I'm inclined to add that.

Reply:
we can also adjust that with the settings.